Source/core/DataElement.h

```diff
@@ -371,15 +438,19 @@ POP_WARNING()
 
             bool expanded = false;
 
-            // Make sure we are not shrinking beyond the size boundary
-            ASSERT(offset <= m_Size);
+            // Make sure we are not expanding beyond the size boundary
+            ASSERT(size <= (m_MaxSize - m_Size));
+
+            if ((expanded = Size(m_Size + size)) == true) {
```

Comment:
This is, I guess a failing testcase. Could you shortly refer to this failing test-case using a JIRA ticket or describe the failing testcase here.

Reply:
@Pierre, I appreciate the feedback and will make the appropriated changes, including filing the Jira ticket.